cmd/init-osm-controller/init-osm-controller.go

```diff
@@ -0,0 +1,110 @@
+
+var settings = cli.New()
+
+var (
+	flags = pflag.NewFlagSet("init-osm-controller", pflag.ExitOnError)
+	log   = logger.New("osm-controller/main")
+
+	meshName           string // An ID that uniquely identifies an OSM instance
+	kubeConfigFilePath string
+	osmNamespace       string
+
+	fetchMeshConfigMaxRetry      = 5
+	fetchMeshConfigRetryInterval = 5
+)
+
+func init() {
+	if err := logger.SetLogLevel("info"); err != nil {
+		log.Fatal().Err(err).Msg("Error setting log level")
+	}
+
+	flags.StringVar(&meshName, "mesh-name", "", "OSM mesh name")
+	flags.StringVar(&kubeConfigFilePath, "kubeconfig", "", "Path to Kubernetes config file.")
+	flags.StringVar(&osmNamespace, "osm-namespace", "", "Namespace to which OSM belongs to.")
+}
+
+func parseFlags() error {
+	if err := flags.Parse(os.Args); err != nil {
+		return err
+	}
+	_ = flag.CommandLine.Parse([]string{})
+	return nil
+}
+
+func main() {
+	log.Info().Msgf("Starting init-osm-controller %s; %s; %s", version.Version, version.GitCommit, version.BuildDate)
+
+	if err := parseFlags(); err != nil {
+		log.Fatal().Err(err).Msg("Failed parsing cmd line arguments")
+	}
+
+	// Initialize kube config and client
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigFilePath)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Failed creating kube config (kubeconfig=%s)", kubeConfigFilePath)
+	}
+
+	configClient, err := configClientset.NewForConfig(kubeConfig)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Could not access Kubernetes cluster, check kubeconfig.")
+		return
+	}
+
+	var retry int
+	for retry = 0; retry < fetchMeshConfigMaxRetry; retry++ {
+		if _, err := configClient.ConfigV1alpha1().MeshConfigs(settings.Namespace()).Get(context.TODO(), meshConfigName, v1.GetOptions{}); err != nil {
+			// MeshConfig does not exist. Create a default one.
+			log.Info().Msgf("MeshConfig not exist. Trying create the default MeshConfig named %s into namespace %s", meshConfigName, osmNamespace)
+
+			if err := createDefaultMeshConfig(configClient); err != nil {
+				log.Warn().Msgf("Failed creating default MeshConfig. %s. Retry after %d seconds.", err, fetchMeshConfigRetryInterval)
+
+				time.Sleep(time.Duration(fetchMeshConfigRetryInterval) * time.Second)
+			} else {
+				log.Info().Msgf("Default MeshConfig named %s created in %s.", meshConfigName, osmNamespace)
+				break
+			}
+		} else {
+			log.Info().Msgf("MeshConfig resource %s exists in %s. Skip creating MeshConfig", meshConfigName, osmNamespace)
+
+			break
+		}
+	}
+
+	if retry == fetchMeshConfigMaxRetry {
+		log.Fatal().Msgf("Max retry (%d) of fetching or creating MeshConfig reached. Exit", fetchMeshConfigMaxRetry)
+	}
+}
+
+func createDefaultMeshConfig(configClient *configClientset.Clientset) error {
+	// create meshConfig from struct
+	meshConfig := &v1alpha1.MeshConfig{}
+	meshConfig.Kind = "MeshConfig"
+	meshConfig.APIVersion = "config.openservicemesh.io/v1alpha1"
+	meshConfig.ObjectMeta.Name = "osm-mesh-config"
```

Comment:
This can be filled in L103 with the struct initialization. Other defaults should go here as well.